src/application/models/v2/Appointments_model_v2.php

```diff
@@ -318,31 +318,31 @@ public function getAllAppointmentBy($service, $aggregates = FALSE, $startDate, $
             $appointments = $this->db->get_where('ea_appointments', $condition)->result_array();
         }
 
-        $maleBookingCounter = 0;
-        $femaleBookingCounter = 0;
-        $undefinedGenderBooking = 0;
         if ($aggregates) {
             foreach ($appointments as &$appointment) {
                 $appointment = $this->get_aggregates($appointment);
-                $gender = $appointment['customer']['gender'];
-                switch(strtoupper($gender)){
-                    case 'M': $maleAppointment++;
-                        break;
-                    case 'F': $femaleAppointment++;
-                        break;
-                    default: $undefinedGenderBooking++;
-                }
             }
         }
+        return $appointments;
+    }
 
-        $statistic['maleBooking'] =  $maleBookingCounter;
-        $statistic['femaleBooking'] =  $femaleBookingCounter;
-        $statistic['undefinedGenderBooking'] =  $undefinedGenderBooking;
-
-        $resultSet['appointments'] = $appointments;
-        $resultSet['statistic'] = $statistic;
-
-        return $resultSet;
+    public function getStatisticAppointment($id_service,$startDate, $endDate){
+        if(strlen($startDate) != 0){
+            $condition['ea_appointments.start_datetime >='] = $startDate;
+        }
+        if(strlen($endDate) != 0){
+            $endDate .= ' 23:59:00';
+            $condition['ea_appointments.end_datetime <='] = $endDate;
+        }
+        $condition["ea_appointments.id_services"] = $id_service;
+
+        $this->db->select('ea_users.gender as Gender, COUNT(ea_appointments.id) as Total')
```

Comment:
make sure the first letter of field name in API json is lowercase

Reply:
updated